src/models/constraint.rs

```diff
@@ -31,3 +40,63 @@ impl Constraint {
     Self { matcher, queries }
   }
 }
+
+// Implements instance methods related to applying a constraint
+impl SourceCodeUnit {
+  pub(crate) fn is_satisfied(
```

Comment:
Ok, other than `pub(crate)`, this is the same code that was in `src_code_unit.rs` before, already under `impl SourceCodeUnit`, right? We are basically moving stuff from an object-oriented structure to an aspect-oriented structure (not exactly, but so to speak). Is my understanding here correct?

Reply:
yes your understanding is correct. It is aspect oriented in some sense (Cross cutting and separation of concern). Actually that was kinda in my mind when i was doing this refactoring. Just wanted to make the main `source_code_unit` less crowded